### PR TITLE
8324756: Test vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize is too slow due to dependency verification

### PR DESCRIPTION
**Pull request by ianrichr**
This change removes dependency verification by passing -XX:-VerifyDependencies in the test.

`vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java` takes 20min to run on linux-x86_64-server-fastdebug:
```
time CONF=linux-x86_64-server-fastdebug make test TEST=vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java
CONF=linux-x86_64-server-fastdebug make test   **1412.82s user 15.27s system 115% cpu 20:41.19 total**
```

Passing -XX:-VerifyDependencies flag speeds up the run time to 1min:
```
time CONF=linux-x86_64-server-fastdebug make test TEST=vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java TEST_VM_OPTS="-XX:-VerifyDependencies"
CONF=linux-x86_64-server-fastdebug make test    **287.27s user 16.19s system 496% cpu 1:01.10 total**
```

Adding -XX:-VerifyDependencies to the test file accomplishes the same run time of 1min:
```
time CONF=linux-x86_64-server-fastdebug make test TEST=vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java
CONF=linux-x86_64-server-fastdebug make test   **272.33s user 14.56s system 464% cpu 1:01.75 total**
```

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [ ] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324756](https://bugs.openjdk.org/browse/JDK-8324756): Test vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize is too slow due to dependency verification (**Bug** - P5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19040/head:pull/19040` \
`$ git checkout pull/19040`

Update a local copy of the PR: \
`$ git checkout pull/19040` \
`$ git pull https://git.openjdk.org/jdk.git pull/19040/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19040`

View PR using the GUI difftool: \
`$ git pr show -t 19040`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19040.diff">https://git.openjdk.org/jdk/pull/19040.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19040#issuecomment-2088848437)

### CODE REVIEW

**Comment by shipilev** [comment]
I think you want to add the reversal of https://github.com/openjdk/jdk/commit/2564f0f99866c33d14947609c276a421ce8cc0a2 to this PR as well.

I am not sure we want to run the test with disabled dependency verification, though. It is a compiler test, so we would like to have compiler checking code online as much as possible. Have you explored if this is an issue with Sweeper removal, and if so, if adding GCs help?

**Comment by ianrichr** [comment]
@shipilev I have experimented with adding a periodic GC (every 5 seconds) in a new thread, and it did not affect the run time of the test. It still timed out at `CONF=linux-x86_64-server-fastdebug make test 1371.53s user 14.98s system 112% cpu 20:31.41 total` with the removal of the `-XX:-VerifyDependencies` flag.

I will submit an amended commit with this test removed from the ProblemList.txt.

File: test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java [hunk 1]
```diff
@@ -163,7 +165,7 @@ protected boolean runThread(int threadNum) throws Throwable {
 
             MHTransformationGen.createAndCallSequence(retVal, dataSnapshot, _mh, _finalArgs, true);
         }
-
+
```
Comment (eme64) [inline test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/stress/compiler/deoptimize/Test.java:168]:
```suggestion

```
Drive by suggestion to get the `rfr` label back ;)

**Comment by eme64** [comment]
Why is `VerifyDependencies` so slow in this test?

**Pull request closed without merging**